contracts/src/types/Selector.sol

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2025 RISC Zero, Inc.
+//
+// All rights reserved.
+pragma solidity ^0.8.20;
+
+using SelectorLib for Selector global;
+using SelectorsLib for Selectors global;
+
+/// @title Selector - A type for handling optional bytes4 values
+/// @notice Provides a safe way to handle optional values in Solidity
+struct Selector {
+    /// @notice The wrapped value, only valid when isSome is true
+    bytes4 value;
+    /// @notice Indicates whether the Option contains a value (true) or is None (false)
+    bool isSome;
+}
+
+/// @title Selectors - A sparse representation of bytes4 selectors
+/// @notice Efficiently stores selectors when most are zero
+struct Selectors {
+    /// @notice Indices where selectors are required
+    uint8[] indices;
+    /// @notice The actual required selectors, corresponding to indices
+    bytes4[] values;
+}
+
+/// @title SelectorLib - Helper library for working with Selector types
+/// @notice Provides utility functions for creating and manipulating Selector values
+library SelectorLib {
```

Comment:
Are the utility functions here and in SelectorsLib just for tests? 

If so we should probably just move them to tests files just to reduce boilerplate in the main code base. Also not sure if the compiler is smart enough to remove unused code so wondering if its inflating bytecode size?

Reply:
Yeah I've dropped all of it and added to TestUtils